src/include/common/concurrent_bitmap.h

```diff
@@ -88,6 +88,78 @@ class RawConcurrentBitmap {
     return false;
   }
 
+  /**
+   * Returns the position of the first unset bit, if it exists.
+   * Note that this result is immediately stale.
+   * @param num_bits number of bits in the bitmap.
+   * @param start_pos start searching from this bit location.
+   * @param[out] out_pos the position of the first unset bit will be written here, if it exists.
+   * @return true if an unset bit was found, and false otherwise.
+   */
+  bool FirstUnsetPos(uint32_t num_bits, uint32_t start_pos, uint32_t *out_pos) {
+    uint32_t num_bytes = BitmapSize(num_bits);  // maximum number of bytes in the bitmap
+    uint32_t byte_pos = start_pos / BYTE_SIZE;  // current byte position
+    uint32_t search_width;                      // number of bits searched at a time
+
+    // optimization: don't search wider than the number of bits we have
+    if (num_bits <= 8) {
```

Comment:
nit: Can you get rid of these magic constants? (Use sizeof(uint8_t), etc. I know this sounds redundant.)